src/jit/liveness.cpp

```diff
@@ -1103,6 +1103,13 @@ void Compiler::fgExtendDbgLifetimes()
             {
                 continue;
             }
+#if !defined(_TARGET_64BIT_) && !defined(LEGACY_BACKEND)
+            // On x86, this applies to long-typed variables as well once we've hit LIR. 
```

Comment:
> We might have to rethink this sort of decomposition after all.

Fixing the liveness issue with the current implementation would have required quite a bit more surgery in order to be anything more than a hack. After some discussion with @CarolEidt, I've updated this change to leave `st.lclVar` nodes that write to a long-typed local as-is. The IR for such nodes now looks like the following after decomposition:

```
         /--*  t0 int
         +--*  t1 int
    t2 = *  gt_long long

         /--*  t2
         *  st.lclVar long V00
```

This construct is then handled in code generation.

If we stick with this formulation in the long term (rather than doing the work to enable recursively promoting vars that represent long-typed fields of promoted struct vars), we should consider leaving sources for such stores undecomposed where possible (e.g. when the source is also an unpromoted long-typed variable or a long-typed field).

@CarolEidt @sivarv @adiaaida @dotnet/jit-contrib PTAL.